src/service/models.py

```diff
@@ -0,0 +1,83 @@
+"""
+Pydantic models for the MinIO Manager API.
+"""
+
+from typing import Annotated, Optional, List
+
+from pydantic import BaseModel, Field
+
+
+class ErrorResponse(BaseModel):
+    """Standard error response model."""
+
+    error: Annotated[int | None, Field(description="Error code")] = None
+    error_type: Annotated[str | None, Field(description="Error type")] = None
+    message: Annotated[str | None, Field(description="Error message")] = None
+
+
+class HealthResponse(BaseModel):
+    """Health check response model."""
+
+    status: Annotated[str, Field(description="Health status")]
+
+
+class UserRequest(BaseModel):
+    """Request model for creating a new user."""
+    
+    groups: Annotated[Optional[List[str]], Field(description="List of groups the user belongs to", default=None)]
+
+
+class UserResponse(BaseModel):
+    """Response model for user operations."""
+    
+    username: Annotated[str, Field(description="Username")]
+    access_key: Annotated[str, Field(description="MinIO access key")]
+    secret_key: Annotated[str, Field(description="MinIO secret key")]
+    policy_name: Annotated[str, Field(description="MinIO policy name")]
+    groups: Annotated[List[str], Field(description="List of groups the user belongs to")]
+    home_path: Annotated[str, Field(description="User's home path in MinIO")]
+    status: Annotated[str, Field(description="User status")]
+
+
+class UserCredentials(BaseModel):
+    """Response model for user credentials."""
+    
+    username: Annotated[str, Field(description="Username")]
+    access_key: Annotated[str, Field(description="MinIO access key")]
+    secret_key: Annotated[str, Field(description="MinIO secret key")]
+    minio_endpoint: Annotated[str, Field(description="MinIO endpoint URL")]
+    home_path: Annotated[str, Field(description="User's home path in MinIO")]
```

Comment:
This seems pretty redundant to the above, but without usage I'm not sure how to evaluate

Reply:
I was thinking one is for user's request and the other one is for admin's request.

Reply:
Ok. You could make a model with the common fields and inherit to DRY things up but I'm not religious

Reply:
yea - I don't have too many models that could benefit from inheriting - I think it maybe fine for now. I can refactor it easily once I have a lot of shared parent fields.